pkg/kube/kube.go

```diff
@@ -783,32 +783,39 @@ var ErrNotFound = errors.New("not found")
 // tries once if wait is zero).
 func (m *kubePackage) kubeGet(ctx context.Context, r *apiResource, wait time.Duration) (runtime.Object, error) {
 	url := m.Master + r.PathWithName()
-	retryCh := make(chan interface{}, 1)
-	retryCh <- struct{}{} // Seed the channel so that we don't wait initially.
 	var waitDone <-chan time.Time
 	if wait != 0 {
 		waitDone = time.After(wait)
 	}
 
+	obj, ok, err := m.kubePeek(ctx, url)
+	if err != nil {
+		return nil, err
+	}
+	if ok {
+		return obj, nil
+	}
+	if waitDone == nil {
+		return nil, ErrNotFound
+	}
+
 	for {
 		select {
-		case retryCh <- time.After(waitRetryInterval):
-		case <-retryCh:
+		case <-time.After(waitRetryInterval):
```

Comment:
How about you use a local variable `w` here, that's initialized with 0 and then set `w = waitRetryInterval` at the end of this case? then you don't need the `kubePeek` logic twice

Reply:
good idea! done